src/node/BrowserFetcher.ts

```diff
@@ -198,21 +199,26 @@ export class BrowserFetcher {
       options.path ||
       path.join(projectRoot, browserConfig[this._product].destination);
     this._downloadHost = options.host || browserConfig[this._product].host;
-    this.setPlatform(options.platform);
+    this.setPlatform(options.platform, this._product);
     assert(
       downloadURLs[this._product][this._platform],
       'Unsupported platform: ' + this._platform
     );
   }
 
-  private setPlatform(platformFromOptions?: Platform): void {
+  private setPlatform(
+    platformFromOptions?: Platform,
+    productFromOptions?: Product
+  ): void {
     if (platformFromOptions) {
       this._platform = platformFromOptions;
       return;
     }
 
     const platform = os.platform();
-    if (platform === 'darwin') this._platform = 'mac';
+    if (platform === 'darwin' && productFromOptions === 'chrome')
+      this._platform = os.arch() === 'arm64' ? 'mac_arm' : 'mac';
+    else if (productFromOptions === 'firefox') this._platform = 'mac';
```

Comment:
it doesn't feel right for me, but firefox ships universal build, so in `downloadURLs` we can't use the separate url for firefox for MacOS ARM because of string interpolation

Reply:
as workaround we can pass platform as query parameter in firefox case and remove this hack